contributed_definitions/nyaml/NXfit_repeated_parameter.yaml

```diff
@@ -0,0 +1,110 @@
+category: base
+doc: |
+  A repeated parameter for a fit function.
+  This would typically be the independent variable in a fit.
```

Comment:
I don't understand what you have in mind here. Does this relate to e.g. many similar peaks with each a set of parameters? Then I don't understand why these would be the independent variable. The fit function _depends_ on the independent variables, parameters are the ones to be determined.

y = Gauss(x, µ, sigma):
y: dependent variable
x: independent variable
µ, sigma: parameters

Reply:
https://en.wikipedia.org/wiki/Dependent_and_independent_variables

Reply:
Here, I tried to go along the lines of how @domna defined [NXdispersion_function](https://fairmat-nfdi.github.io/nexus_definitions/classes/contributed_definitions/NXdispersion_function.html#nxdispersion-function). I guess a `repeated_parameter` is meant as one that changes for each point in the independent variable. Maybe @domna can say what the different between `single` and `repeated` parameters is?

Reply:
It's more meant to structure a function with multiple peaks. In dispersions you often have, e.g., a background susceptibility and on top you have some Lorentz peaks or so. The single parameter is meant to catch the parameter which exists only once for the model, the repeated parameter is meant to carry the information of each repeating peak (with the same parameter set). The total function is then the sum of the single parameter and repeated parameter functions.

In dispersions we need this for, e.g., the Tauc Lorentz model which has a band gap parameter for the whole model but multiple oscillators with the same parameter set. The band gap would be a single parameter and the oscillator parameter set the repeating ones.

Reply:
For the formula I have this `sum[...]` part which indicates that everything in there is the repeating part of the function, see this explanation https://fairmat-nfdi.github.io/nexus_definitions/ellipsometry-structure.html#base-classes

Reply:
In what I was envisioning for XPS, the peaks (and background) are often using a different function with different parameters, so the idea of a repeated parameter doesn't apply here.

In XPS, the peaks are defined independently (with their own parameter set) and then the resulting fit is a of the form `model = peak1 + peak2 + backgr. So my idea was, in one `NXfit` to give each `NXpeak` its own `NXfit_function` (each with a set of single parameters) and then for the global fit, you add another `NXfit_function` that e.g. does a least linear fit of the sum given above. But I am not so sure how to implement this and if it's general enough. Any ideas?